src/wp-includes/class-wp-query.php

```diff
@@ -3257,6 +3257,8 @@ public function get_posts() {
 				$post_ids[]                      = (int) $post->ID;
 			}
 
+			wp_cache_add_multiple( $post_parents, 'post_parent' );
```

Comment:
This post parent is cache is only used here for now. It might be useful in other place. But now, I have avoided priming this cache in `update_post_cache`, as in most use case, this cache is not useful. Priming would result in using session memory for a cache that is not used.

Reply:
Only save the cache if needed. For sites that do no use id=>parent much, the caches will not be primed, which is a good thing.

Reply:
An inline comment that documents what you've noted here could be useful for future reference.